site/content/blog/2025-02-17-portswigger-lab-walkthrough-broken-brute-force-protection-ip-block/index.md

```diff
@@ -0,0 +1,755 @@
+
+![move-message-processor-script-top.png](images/move-message-processor-script-top.png)
+
+You can optionally move the script to the top of the processor list. Then, run the Fuzzer.
+
+## Have Username, Got Password
+
+Once the Fuzzer is done, we can inspect the results. We’re looking for indications of a successful login. So, we’ll sort the results in descending order using the status code column (simply “*Code”* in the UI).
+
+> [!NOTE]
+> Both of these solutions are valid. As you might have noticed if you’ve solved a lab more than once, different lab instances can have different password solutions. So, you could see different passwords than shown below.
+
+### Method 1: Interleave Wordlists With Valid Credentials
+
+Our sorted results from our bypass with valid credentials show multiple `302 Found` results. This is because of the logins to our own account. So, we want to search the *Payloads* column for a payload that includes “carlos”:
+
+![302-found-carlos.png](images/302-found-carlos.png)
+
+Voila! Carlos’ password is “soccer”. Log in with the password to solve the lab.
+
+### Method 2: Wait Out the IP Block
+
+For the second method, we also want to sort the results in descending order by status code. At the top of our sorted results is a `302 Found`:
+
+![password-found.png](images/password-found.png)
+
+This is a different lab instance than used with the bypass method. The password here is “biteme”, as we can see either through the *Payloads* column or in the request body:
+
+![lab-solved.png](images/lab-solved.png)
+
+We’ll confirm by logging in to Carlos’ account.
+
+## Takeaway
+
+Rate-limiting with IP blocking is a common defense against brute-force attacks. But, it can be bypassed if implemented poorly. Understanding how attackers can exploit logic flaws to bypass security measures can help developers build more secure systems. Explore other labs to learn additional ways attackers can exploit authentication mechanisms and workflows. You can also find more [PortSwigger lab walkthroughs](https://www.zaproxy.org/tags/portswigger-lab/) using ZAP in the Docs.
+
+## Resources
+
+1. [Authentication vulnerabilities](https://portswigger.net/web-security/authentication)
+2. [Burp to ZAP Feature Map](https://www.zaproxy.org/docs/burp-to-zap-feature-map/)
+3. [PortSwigger Lab Walkthroughs With ZAP](https://www.zaproxy.org/tags/portswigger-lab/)
+4. [ZAP Docs](https://www.zaproxy.org/docs/)
+5. [Migration Guide from Nashorn to GraalJS](https://www.graalvm.org/latest/reference-manual/js/NashornMigrationGuide/)
+6. [GraalJS Compatibility](https://www.graalvm.org/latest/reference-manual/js/JavaScriptCompatibility/)
+7. [Java Interoperability](https://www.graalvm.org/jdk21/reference-manual/js/JavaInteroperability/)
+
+_Image credit: [Pexels](https://www.pexels.com/photo/real-estate-concept-with-key-and-house-models-31424880/)._
```

Comment:
I needed to include some attribution but wasn't sure where to add it. Is this okay?

Oh, and what do you think of the image itself?

Reply:
I'm good with this, and I'm good with the image too.

Reply:
The license says that attribution is not required:
https://www.pexels.com/license/

> Attribution is not required. Giving credit to the photographer or Pexels is not necessary but always appreciated.

Not that I don't mind the attribution just saying.